src/hotspot/cpu/x86/sharedRuntime_x86_64.cpp

```diff
@@ -1744,6 +1744,7 @@ nmethod* SharedRuntime::generate_native_wrapper(MacroAssembler* masm,
                                               in_ByteSize(-1),
                                               oop_maps,
                                               exception_offset);
+    if (nm == nullptr) vm_exit_during_initialization("Initial size of CodeCache is too small");
```

Comment:
IIUC `AdapterHandlerLibrary::create_native_wrapper`  is the only caller of `generate_native_wrapper` so it is only called during VM init now. But not sure it wouldn't be better/cleaner to simply return nullptr here and have `AdapterHandlerLibrary::create_native_wrapper` do the `vm_exit_during_initialization`?

Reply:
We could do that too. We have to check for nullptr && method->is_continuation_native_intrinsic() since AdapterHandlerLibrary::create_native_wrapper() itself is called from 2 other places which should not fail with vm_exit_during_initialization() if creation of the native nmethod failed. I don't have a preference, both ways are fine with me.

Reply:
Sorry for the delay. I've been looking at the proposed initialization sequence:
```
Threads::create_vm
  initialize_class(vmSymbols::jdk_internal_vm_Continuation(), CHECK
    // causes linking of Continuation
    InstanceKlass::link_class_impl(TRAPS)
      InstanceKlass::link_methods(TRAPS)
        Method::link_method(mh, TRAPS)
          AdapterHandlerLibrary::create_native_wrapper
            SharedRuntime::generate_native_wrapper
```
and it seems to me that if `generate_native_wrapper` returns null then `create_native_wrapper` can check for that and also return early. Then `link_method` can do what other callers of `create_native_wrapper` do and simply check the resulting method and throw an exception:
```
if (!h_method->has_compiled_code()) {
  // throw OOME
}
```
that exception will propagate up and terminate the VM the same as any other exception during vm init would.

What do you think?

Reply:
Yes, I like that approach. I changed it in the last commit.